src/common/CMakeLists.txt

```diff
@@ -1,2 +1,21 @@
+# Compiler properties
+
+set(COMMON_TARGET_PROPERTIES 
+  POSITION_INDEPENDENT_CODE ON   # -fPIC
+  C_STANDARD 99                  # -std=gnu99
+  )
+
+# Target TARGET_LIBCOMMON_STATIC
+
 add_library(${TARGET_LIBCOMMON_STATIC} STATIC common.c)
+set_target_properties(${TARGET_LIBCOMMON_STATIC} PROPERTIES ${COMMON_TARGET_PROPERTIES})
+
+# Target common.o
+
+add_library(common_obj OBJECT common.c)
```

Comment:
Why do you need the object here? You should use the `${TARGET_LIBCOMMON_STATIC}` library built on line 10.

Reply:
What I want to do is to add those object files into a shared library. If they are object files, it can be done with $<TARGET_OBJECTS:objlib>, but how can I do this with a static library?

https://stackoverflow.com/questions/11429055/cmake-how-create-a-single-shared-library-from-all-static-libraries-of-subprojec